dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/workflow/internalspi/InternalWorkflowStepSPI.java

```diff
@@ -27,15 +29,24 @@
     private final WorkflowStepRegistry stepRegistry;
     private final DatasetVersionServiceBean datasetVersions;
     private final CitationFactory citationFactory;
+    private final AuthenticationServiceBean authenticationService;
+    private Clock clock;
 
     // -------------------- CONSTRUCTORS --------------------
 
     @Inject
     public InternalWorkflowStepSPI(WorkflowStepRegistry stepRegistry, DatasetVersionServiceBean datasetVersions,
-                                   CitationFactory citationFactory) {
+                                   CitationFactory citationFactory, AuthenticationServiceBean authenticationService) {
```

Comment:
Most probably `mxrdr-extension` will give compile error after this (one of the test is using this constructor). So we probably want to adjust the changes also there.